products/revenue_analytics/backend/hogql_queries/revenue_example_data_warehouse_tables_query_runner.py

```diff
@@ -1,77 +1,74 @@
-from typing import Union
+from typing import cast, Union
 
 from posthog.hogql import ast
 from posthog.hogql.constants import LimitContext
 from posthog.hogql_queries.insights.paginators import HogQLHasMorePaginator
 from posthog.hogql_queries.query_runner import QueryRunner
+from posthog.hogql.database.database import create_hogql_database, Database
+from posthog.hogql.hogql import HogQLContext
 from posthog.schema import (
     RevenueExampleDataWarehouseTablesQuery,
     RevenueExampleDataWarehouseTablesQueryResponse,
     CachedRevenueExampleDataWarehouseTablesQueryResponse,
 )
-
-from posthog.hogql.database.schema.exchange_rate import (
-    DEFAULT_CURRENCY,
-    currency_expression_for_data_warehouse,
-    revenue_expression_for_data_warehouse,
-)
+from ..models import RevenueAnalyticsRevenueView
 
 
 class RevenueExampleDataWarehouseTablesQueryRunner(QueryRunner):
     query: RevenueExampleDataWarehouseTablesQuery
     response: RevenueExampleDataWarehouseTablesQueryResponse
     cached_response: CachedRevenueExampleDataWarehouseTablesQueryResponse
     paginator: HogQLHasMorePaginator
+    database: Database
+    hogql_context: HogQLContext
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.paginator = HogQLHasMorePaginator.from_limit_context(
             limit_context=LimitContext.QUERY, limit=self.query.limit if self.query.limit else None
         )
 
-    def to_query(self) -> Union[ast.SelectQuery, ast.SelectSetQuery]:
-        tracking_config = self.query.revenueTrackingConfig
+        # We create a new context here because we need to access the database
+        # below in the to_query method and creating a database is pretty heavy
+        # so we'll reuse this database for the query once it eventually runs
+        self.database = create_hogql_database(team=self.team)
+        self.hogql_context = HogQLContext(team_id=self.team.pk, database=self.database)
 
+    def to_query(self) -> Union[ast.SelectQuery, ast.SelectSetQuery]:
         queries = []
-        if tracking_config.dataWarehouseTables:
-            for table in tracking_config.dataWarehouseTables:
+
+        # UNION ALL for all of the `RevenueAnalyticsRevenueView`s
+        for view_name in self.database.get_views():
+            view = self.database.get_table(view_name)
+            if isinstance(view, RevenueAnalyticsRevenueView):
+                view = cast(RevenueAnalyticsRevenueView, view)
```

Comment:
I think this cast is unnecessary now?

Reply:
I'll check. I think it's still needed because mypy is not very smart